### PR TITLE
[skip ci] Allow ModelCheckpoint in Trainer callbacks list

### PR DESCRIPTION
**Pull request by awaelchli**
Allows passing ModelCheckpoint directly to the callbacks list:

```
Trainer(callbacks=[ModelCheckpoint(...)])
```
This was not possible before because **save_function** was not set. We now set it in the callback directly.

The following cases need to be considered:
- `Trainer(checkpoint_callback=True, callbacks=[ModelCheckpoint()])`
- `Trainer(checkpoint_callback=False, callbacks=[ModelCheckpoint()])`
- `Trainer(checkpoint_callback=ModelCheckpoint(), callbacks=[ModelCheckpoint()])`

This PR ignores the `checkpoint_callback` setting if one is passed to callbacks. Does this make sense @PyTorchLightning/core-contributors ?

TODO: 

- [x] needs a test
- [x] need to define `trainer.checkpoint_callback`, we probably need to limit it to 1 checkpoint callback for now?

### CODE REVIEW

**Comment by rohitgr7** [comment]
Seems good. Should we add a warning too if both are provided since this happens during initialization? One can change it if required.

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 1]
```diff
@@ -44,12 +44,17 @@ def on_trainer_init(
         )
 
     def init_default_checkpoint_callback(self, checkpoint_callback):
-        if checkpoint_callback is True:
+        via_callbacks = any(isinstance(c, ModelCheckpoint) for c in self.trainer.callbacks)
+        if via_callbacks:
+            rank_zero_warn(
+                "Callbacks list contains a ModelCheckpoint instance."
+                " Trainer(checkpoint_callback=...) will be ignored."
+            )
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/connectors/callback_connector.py:52]: maybe do this working if checkpoint_callback is True or is an instance of ModelCheckpoint.

Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/callback_connector.py:52]: what if user has checkpoint_callback=False? isn't it confusing to pass checkpoint callback and it still does not turn it off?

Comment (rohitgr7) [reply pytorch_lightning/trainer/connectors/callback_connector.py:52]: I wonder why anyone would do that.. But good point :]

**Comment by awaelchli** [comment]
The tests currently fail as they reveal another bug, which I will fix here: #4027

**Pull request closed without merging**